src/nsidc/metgen/metgen.py

```diff
@@ -361,13 +345,10 @@ def null_operation(configuration: config.Config, granule: Granule) -> Granule:
 
 def granule_collection(configuration: config.Config, granule: Granule) -> Granule:
```

Comment:
If we get this decorator from the Python stdlib:
`from functools import cache`
then we can add the decorator to the line before the function definition:
`@cache`
This will magically turn the function into one that only gets executed once when called multiple times with the same parameter. 

This would allow us to update the guts of the `granule_collection` function to call CMR, but it will only ever execute once, even though it gets called for each granule. By adding this decorator we can keep the rest of the code as-is.

Reply:
Here's the docs for `cache`. https://docs.python.org/3/library/functools.html#functools.cache

Reply:
It's kind of a neat trick.

Reply:
I'll give it a try!